src/hooks.js

```diff
@@ -0,0 +1,41 @@
+import { useState, useEffect } from 'preact/hooks'
+
+const fetchBody = async (url, fetchOpts) => {
+  const state = {}
+  try {
+    state.response = await fetch(url, fetchOpts)
+    state.body = await state.response.text()
```

Comment:
I take the body as text and not as JSON directly for 2 reasons:
- We can use it for fetching things that are not JSON
- If we get a malformated JSON we can also see the orignal value, usefull for debugs